core/src/test/scala/org/apache/spark/sql/delta/MergeIntoMaterializeSourceSuite.scala

```diff
@@ -0,0 +1,279 @@
+/*
+ * Copyright (2021) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.delta
+
+import scala.collection.mutable
+import scala.util.control.NonFatal
+
+import org.apache.spark.sql.delta.DeltaTestUtils._
+import org.apache.spark.sql.delta.cdc.MergeCDCSuite
+import org.apache.spark.sql.delta.commands.MergeStats
+import org.apache.spark.sql.delta.commands.merge.{MergeIntoMaterializeSourceError, MergeIntoMaterializeSourceErrorType, MergeIntoMaterializeSourceReason}
+import org.apache.spark.sql.delta.commands.merge.MergeIntoMaterializeSource.mergeMaterializedSourceRddBlockLostErrorRegex
+import org.apache.spark.sql.delta.sources.DeltaSQLConf
+import org.apache.spark.sql.delta.util.JsonUtils
+import org.scalactic.source.Position
+import org.scalatest.Tag
+
+import org.apache.spark.SparkException
+import org.apache.spark.sql.{DataFrame, Row}
+import org.apache.spark.sql.catalyst.expressions.{AttributeReference, EqualTo, Expression, Literal}
+import org.apache.spark.sql.catalyst.plans.logical._
+import org.apache.spark.sql.execution.{FileSourceScanExec, FilterExec, LogicalRDD, RDDScanExec, SQLExecution, WholeStageCodegenExec}
+import org.apache.spark.sql.execution.aggregate.HashAggregateExec
+import org.apache.spark.sql.functions._
+import org.apache.spark.sql.internal.SQLConf
+import org.apache.spark.sql.types._
+import org.apache.spark.storage.StorageLevel
+import org.apache.spark.util.Utils
+
+trait MergeIntoMaterializeSourceTests extends MergeIntoSQLSuite {
```

Comment:
Why are you extending the entire MergeIntoSQLSuite? do we really need to rerun ALL the tests in the MergeIntoSQLSuite to test the materialization?

MergeInto***Suite is one of longest running test suite and I hate to duplicate test suite. Our test run times are already multiple hours.